docs/devel/kubectl-conventions.md

```diff
@@ -135,6 +133,151 @@ Updated: 8/27/2015
 * Use "TYPE" for the particular flavor of resource type accepted by kubectl, rather than "RESOURCE" or "KIND"
 * Use "NAME" for resource names
 
+## Command implementation conventions
+
+For every command there should be a `NewCmd<CommandName>` function that creates the command and returns a pointer to a `cobra.Command`, which can later be added to other parent commands to compose the structure tree. There should also be a `<CommandName>Options` struct with a variable to every flag and argument declared by the command (and any other variable required for the command to run). This makes tests and mocking easier. The struct ideally exposes three functions:
```

Comment:
Options in `<CommandName>Options` struct name makes impression it contains only flags. I.e. all values that are passed down by `cobra.Command`. As the paragraph goes on it says: "with a variable and every flag ... (`and any other variable required for the command to run`)". Any other variable required for the command to run does not come under `Options`. It is more configuration related. It would make more sense to use `<CommandName>Config` instead. Options are part of configuration so as other variables like client, factory, mapper, etc.